blockchain/chain/src/store/chain_store.rs

```diff
@@ -560,6 +534,33 @@ where
         Ok(())
     }
 
+    pub fn sub_head_changes(&self) -> Receiver<HeadChange> {
+        let (tx, rx) = bounded(16);
+        let mut subscriber = self.publisher.subscribe();
+
```

Comment:
You need to get the current heaviest tipset here and send it through `tx`. ChainNotify states that the first message should be a HeadChange::Current. https://pkg.go.dev/github.com/filecoin-project/lotus/api#FullNode.

Reply:
oops, missed that detail